codec-http2/src/main/java/io/netty/handler/codec/http2/Http2FrameCodec.java

```diff
@@ -231,7 +231,7 @@ void handlerAdded0(@SuppressWarnings("unsed") ChannelHandlerContext ctx) throws
      */
     @Override
     public final void userEventTriggered(ChannelHandlerContext ctx, Object evt) throws Exception {
-        if (evt instanceof Http2ConnectionPrefaceWrittenEvent) {
+        if (evt == Http2ConnectionPrefaceAndSettingsFrameWrittenEvent.INSTANCE) {
```

Comment:
lets be consistent and either use `evt == Http2ConnectionPrefaceAndSettingsFrameWrittenEvent.INSTANCE` or `instanceof` ... `instanceof` allows us to be less dependent on the class hierarchy but I think either way will work.

Reply:
@lionelli1 @Scottmitch I vote for using `==` here. @lionelli1 please make the change and squash. After this is done i will merge it.

Reply:
ok! replaced all `instanceof` with `==` and squashed my changes. thanks!